docs/development/versioning.md

```diff
@@ -0,0 +1,74 @@
+# Versioning
+
+## Version changes
+
+Polars adheres to the [semantic versioning](https://semver.org/) specification.
+
+As Polars has not released its `1.0.0` version yet, breaking releases lead to a minor version increase (e.g. from `0.18.15` to `0.19.0`), while all other releases increment the patch version (e.g. from `0.18.14` to `0.18.15`)
+
+## Policy for breaking changes
+
+Polars takes backwards compatibility seriously, but we are not afraid to change things if it leads to a better product.
+
+!!! warning Rust users only
+
+    The Rust API for Polars is currently not considered stable.
+    Functionality can be changed or removed without warning.
+
+### Philosophy
+
+We don't always get it right on the first try.
+We learn as we go along and get feedback from our users.
+Sometimes, we're a little too eager to get out a new feature and didn't ponder all the possible implications.
+
+If this happens, we correct our mistakes and introduce a breaking change.
+Most of the time, this is no big deal.
+Users get a deprecation warning, they do a quick search-and-replace in their code base, and that's that.
+
+At times, we run into an issue requires more effort on our user's part to fix.
+A change in the query engine can seriously impact the assumptions in a data pipeline.
+We do not make such changes lightly, but we will make them if we believe it makes Polars better.
+
+Freeing ourselves of past indiscretions is important to keep Polars moving forward.
+We know it takes time and energy for our users to keep up with new releases, but in the end, it benefits everyone for Polars to be the best product possible.
```

Comment:
We know it takes time and energy for our users to keep up with new releases but, in the end, it benefits everyone for Polars to be the best product possible.